### PR TITLE
tests: posix: Function call return values need to be validated.

### LINKED ISSUE #7084: [Coverity CID: 185280] Error handling issues in /tests/posix/pthread_rwlock/src/posix_rwlock.c
Static code scan issues seen in File: /tests/posix/pthread_rwlock/src/posix_rwlock.c
 Category: Error handling issues
 Function: test_rw_lock
 Component: Tests
 CID: 185280
 Please fix or provide comments to square it off in coverity in the link: https://scan9.coverity.com/reports.htm#v32951/p12996

### PR DESCRIPTION
**Pull request by sritej20**
In this fix we are checking the return value of pthread_attr_init before
asserting a pthread_attr_destroy and init retry.

Resolves #7084
Coverity-CID: 185280

Signed-off-by: Sritej Kanakadandi Venkata Rama <sritej.kvr@gmail.com>

### CODE REVIEW

File: tests/posix/pthread_rwlock/src/posix_rwlock.c [hunk 1]
```diff
@@ -80,8 +80,14 @@ static void test_rw_lock(void)
...
 	/* Creating N premptive threads in increasing order of priority */
 	for (i = 0; i < N_THR; i++) {
-		pthread_attr_destroy(&attr[i]);
-		pthread_attr_init(&attr[i]);
+
+		ret = pthread_attr_init(&attr[i]);
+		if (ret != 0) {
+			zassert_false(pthread_attr_destroy(&attr[i]),
+					"Unable to destroy pthread object attrib\n");
+			zassert_false(pthread_attr_init(&attr[i]),
```
Comment (lpereira) [inline tests/posix/pthread_rwlock/src/posix_rwlock.c]: What is this trying to accomplish here?  If the call to pthread_attr_init() fails, the attributed is destroyed and initialized again.  It's not clear to me why this is being performed this way.

Also, the zassert_false() calls do not make much sense: both pthread_attr_destroy() and pthread_attr_init() return an integer error code, not a boolean value.

**Review by lpereira — CHANGES_REQUESTED** [review]

**Comment by sritej20** [comment]
@lpereira made the changes you requested using the merge resolution. removed the extra destroy and just check the init return value

**Comment by sritej20** [comment]
closing PR as CI is failing due to merge conflicts. Will open a new PR with suggested review comments.

**Comment by lpereira** [comment]
No need to open a new PR.  Just rebase, fix the merge conflicts, and repush to this branch.

**Pull request closed without merging**